zerver/forms.py

```diff
@@ -144,9 +144,6 @@ def clean_email(self):
         if realm is None or realm.invite_required:
             raise ValidationError(mark_safe(SIGNUP_STRING))
 
-        if completely_open(realm.domain):
-            return email
-
```

Comment:
This change I believe eliminates our restricted_to_domain check.  I'm surprised this didn't fail tests, but it's possible we don't have a test for that issue...